base/server/healthcheck/pki/server/healthcheck/clones/plugin.py

```diff
@@ -0,0 +1,185 @@
+            output = e.output.decode('utf-8')
+            return output
+
+        output = output.decode('utf-8')
+
+        return output
+
+    def contact_subsystem_using_sslget(
+            self, port, host, subsystemnick,
+            token_pwd, db_path, params, url):
+
+        command = ["/usr/bin/sslget"]
+
+        if subsystemnick is not None:
+            command.extend(["-n", subsystemnick])
+
+        command.extend(["-p", token_pwd, "-d", db_path])
+
+        if params is not None:
+            command.extend(["-e", params])
+
+        command.extend([
+            "-r", url, host + ":" + port])
+
+        logger.info(' command : %s ', command)
+        output = None
+        try:
+            output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode('utf-8')
+            return output
+
+        output = output.decode('utf-8')
+
+        return output
+
+    def get_security_domain_data(self, host, port):
+        domain_data = None
+
+        try:
+            connection = PKIConnection(protocol='http',
+                                       hostname=host,
+                                       port=port,
+                                       verify=False)
+
+            securityDomainClient = SecurityDomainClient(connection)
+            domain_data = securityDomainClient.get_domain_info()
+
+        except BaseException as e:
+            logger.error("Internal server error %s", e)
+            return domain_data
+
+        systems = domain_data.subsystems
+        for s in systems.values():
+            for h in s.hosts.values():
+                if s.id == 'CA':
+                    if h.Clone == 'TRUE':
+                        self.clone_cas.append(h)
+                    else:
+                        self.master_cas.append(h)
+                elif s.id == 'KRA':
+                    if h.Clone == 'TRUE':
+                        self.clone_kras.append(h)
+                    else:
+                        self.master_kras.append(h)
+                elif s.id == 'OCSP':
+                    if h.Clone == 'TRUE':
+                        self.clone_ocsps.append(h)
+                    else:
+                        self.master_ocsps.append(h)
+                elif s.id == 'TPS':
+                    if h.Clone == 'TRUE':
+                        self.clone_tpss.append(h)
+                    else:
+                        self.master_tpss.append(h)
+                elif s.id == 'TKS':
+                    if h.Clone == 'TRUE':
+                        self.clone_tkss.append(h)
+                    else:
+                        self.master_tkss.append(h)
+
+        return domain_data
+
+    def get_security_domain_ca(self):
+        sec_domain = None
+        sechost = None
+        secport = None
+        ca_subsystem = self.instance.get_subsystem('ca')
+        if(ca_subsystem):
```

Comment:
(nitpick) no brackets. Not sure if the future lint rule would complain, but just in case :-)